python/api/client.py

```diff
@@ -0,0 +1,514 @@
+    api_token: str
+    base_url: str | None = None
+
+
+class ApiClient:
+    def __init__(self, config: ApiConfig):
+        self.config = config
+        self.base_url = config.base_url or BASE_URL
+        self.client = httpx.AsyncClient()
+
+    def _build_headers(self) -> dict[str, str]:
+        return {
+            "Authorization": f"Bearer {self.config.api_token}",
+            "Content-Type": "application/json",
+        }
+
+    async def _fetch_with_schema(
+        self,
+        url: str,
+        response_class: type[T],
+        method: str = "GET",
+        data: dict[str, Any] | None = None,
+        timeout: float = 15.0,
+    ) -> Result[T]:
+        try:
+            headers = self._build_headers()
+
+            if method == "GET":
+                response = await self.client.get(url, headers=headers, timeout=timeout)
+            elif method == "POST":
+                response = await self.client.post(url, headers=headers, content=json.dumps(data) if data else None, timeout=timeout)
+            elif method == "PATCH":
+                response = await self.client.patch(url, headers=headers, content=json.dumps(data) if data else None, timeout=timeout)
+            elif method == "DELETE":
+                response = await self.client.delete(url, headers=headers, timeout=timeout)
+            else:
+                raise ValueError(f"Unsupported HTTP method: {method}")
+
+            if not response.is_success:
+                if response.status_code == 401:
+                    raise Exception(ErrorCode.INVALID_API_KEY)
+
+                try:
+                    error_data = response.json()
+                    if error_data.get("type") == "validation_error" and error_data.get("code"):
+                        raise Exception(f"Validation error: {error_data['code']}")
+                except Exception:
+                    pass
+
+                raise Exception(f"Request failed: {response.text}")
+
+            raw_data = response.json()
+
+            try:
+                validated_data = response_class.model_validate(raw_data)
+                return SuccessResult(validated_data)
+            except ValidationError as e:
+                raise Exception(f"Response validation failed: {e}") from e
+
+        except Exception as error:
+            return ErrorResult(error)
+
+    def organizations(self):
+        return OrganizationResource(self)
+
+    def projects(self):
+        return ProjectResource(self)
+
+    def feature_flags(self, project_id: str):
+        return FeatureFlagResource(self, project_id)
+
+    def insights(self, project_id: str):
+        return InsightResource(self, project_id)
+
+    def dashboards(self, project_id: str):
+        return DashboardResource(self, project_id)
+
+    def query(self, project_id: str):
+        return QueryResource(self, project_id)
+
+    def users(self):
+        return UserResource(self)
+
+    async def close(self):
+        await self.client.aclose()
+
+
```

Comment:
Would it also make sense to add this in case we forget to call close manually?

```suggestion
      async def __aexit__(self, exc_type, exc_val, exc_tb):
          await self.client.aclose()
          return False

```

Reply:
yeah good call - I've added support for this as a context manager - not yet sure if we'll use that in any of the integrations but I'll see